src/inmanta/deploy/scheduler.py

```diff
@@ -360,6 +372,28 @@ async def get_facts(self, resource: dict[str, object]) -> None:
             )
         )
 
+    async def trigger_deploy_for_resource(self, resource: ResourceIdStr, reason: str, priority: TaskPriority) -> None:
+        """
+        Make sure the given resource is marked for deployment with at least the provided priority.
+        If the given priority is higher than the previous one (or if it didn't exist before), a new deploy
+        task will be scheduled with the provided reason.
+        """
+        async with self._scheduler_lock:
+            if resource not in self._state.resource_state:
+                # The resource was removed from the model by the time this method was triggered
+                return
+
+            if self._state.resource_state[resource].blocked is BlockedStatus.YES:  # Can't deploy
+                return
+            to_deploy: set[ResourceIdStr] = {resource}
+
+            self._work.deploy_with_context(
```

Comment:
I think it would be good to also remove the timer in this method. In the current implementation the timer manager is the only user of this method, so it will be a NOOP (and admittedly a bit strange) except that it cleans up the timer instance. But if we ever call this from the outside, we want the timer to disappear.